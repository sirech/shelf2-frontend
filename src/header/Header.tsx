import React, { useState } from 'react'
import { Link, useNavigate } from 'react-router-dom'

import {
  Collapse,
  Nav,
  Navbar,
  NavbarBrand,
  NavItem,
  NavbarToggler,
} from 'reactstrap'
import Logout from './Logout'
import SearchBar from './SearchBar'

const Login = () => (
  <NavItem>
    <Link to="/login" className="nav-link">
      Login
    </Link>
  </NavItem>
)

const Header = () => {
  const [isOpen, setIsOpen] = useState(false)
  const navigate = useNavigate()
  return (
    <header className="mb-4">
      <Navbar dark color="dark" expand="sm">
        <NavbarBrand onClick={() => navigate('/books')}>Shelf</NavbarBrand>
        <NavbarToggler onClick={() => setIsOpen(!isOpen)} />
        <Collapse isOpen={isOpen} navbar>
          <Nav className="ml-auto" navbar>
            <SearchBar />
            <Login />
            <Logout />
          </Nav>
        </Collapse>
      </Navbar>
    </header>
  )
}

export default Header
