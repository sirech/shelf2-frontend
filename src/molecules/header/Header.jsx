import React from 'react'
import { Link } from 'react-router-dom'
import { LinkContainer } from 'react-router-bootstrap'

import { Collapse, Nav, Navbar, NavbarBrand, NavItem, NavbarToggler } from 'reactstrap'

class Header extends React.Component {
  constructor (props) {
    super(props)

    this.toggle = this.toggle.bind(this)
    this.state = {
      isOpen: false
    }
  }

  toggle () {
    this.setState({
      isOpen: !this.state.isOpen
    })
  }

  render () {
    return (
      <header className='mb-4'>
        <Navbar inverse color='inverse'>
          <NavbarToggler right onClick={this.toggle} />
          <LinkContainer to='/' >
            <NavbarBrand>Shelf</NavbarBrand>
          </LinkContainer>
          <Collapse isOpen={this.state.isOpen} navbar>
            <Nav className='ml-auto' navbar>
              <NavItem>
                <Link to='/books' className='nav-link'>Books</Link>
              </NavItem>
            </Nav>
          </Collapse>
        </Navbar>
      </header>
    )
  }
}

export default Header
