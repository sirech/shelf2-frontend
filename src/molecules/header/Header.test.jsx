import React from 'react'
import { fullRender } from '../../test'
import Header from './Header'

describe('components', () => {
  describe('Header', () => {
    it('renders correctly', () => {
      const { component } = fullRender(<Header />)
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
