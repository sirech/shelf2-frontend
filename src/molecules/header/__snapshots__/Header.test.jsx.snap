// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Header renders correctly 1`] = `
<header
  className="mb-4"
>
  <nav
    className="navbar navbar-inverse bg-inverse"
    role="navigation"
  >
    <button
      className="navbar-toggler navbar-toggler-right"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <a
      className=" active navbar-brand"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Shelf
    </a>
    <div
      className="collapse navbar-collapse"
      style={Object {}}
    >
      <ul
        className="ml-auto navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/books"
            onClick={[Function]}
          >
            Books
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
